src/main/java/com/xceptance/neodymium/common/EndTestStepListener.java

```diff
@@ -0,0 +1,30 @@
+package com.xceptance.neodymium.common;
+
+import com.codeborne.selenide.logevents.LogEvent;
+import com.codeborne.selenide.logevents.LogEventListener;
+import com.xceptance.neodymium.util.AllureAddons;
+import com.xceptance.neodymium.util.Neodymium;
+
+public class EndTestStepListener implements LogEventListener
+{
+    public static final String LISTENER_NAME = "end-teststep-listener";
+
+    private String lastUrl = "";
```

Comment:
I fear we might get into issues with this approach. We add this listener inside the NeodymiumRunner it's a singel instance which will be used for all test cases from that point on, but I'm not sure how this works for multiple running tests, especially in a parallel execution.

To be on the save side we could use the Neodymium data storage to store the last URL. Since this is concurrency safe.

Something Like this: `Neodymium.getData().get("neodymium.data.lastUrl")`